### PR TITLE
Add webring maintenance scripts and GitHub action

### PR DESCRIPTION
**Pull request by KhushrajRathod**
Closes https://github.com/hackclub/webring/issues/43

The main members.json file (the one to be edited by future members) has been moved to the top level. The one inside public/ is generated every 15 minutes.

### CODE REVIEW

**Comment by googol88** [comment]
This really cool!! Just one thing, do you think 15 minutes is too often? Maybe we can change it to once a day instead...
Also I haven't worked with Actions before so I'll wait to see what sam thinks.
Really neat work!

File: .github/workflows/maintenance.yml [hunk 1]
```diff
@@ -27,10 +27,10 @@ jobs:
       run: |
         git config --local user.email "41898282+github-actions[bot]@users.noreply.github.com"
         git config --local user.name "github-actions[bot]"
-        git commit -m "Maintain Webring" -a
+        git commit -m "Maintain Webring" -a --allow-empty
```
Comment (sampoder) [inline .github/workflows/maintenance.yml:30]: Could we remove this? Would reduce spamming the commit history.

**Review by sampoder — CHANGES_REQUESTED** [review]

**Comment by sampoder** [comment]
Otherwise, I'm pretty happy with this.

**Comment by KhushrajRathod** [comment]
@sampoder not sure how but....
Without --allow-empty, I'd have to make it succeed anyway with `|| true`, and that'd also make the push fail (right?).
I'll give it a go

**Review by sampoder — APPROVED** [review]

**Comment by sampoder** [comment]
Perfecto, that seemed to work @KhushrajRathod!

**Comment by googol88** [comment]
Woohoo! This is a really nice addition!

**Pull request merged**